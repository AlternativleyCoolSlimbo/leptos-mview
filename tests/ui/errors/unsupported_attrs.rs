use leptos::*;
use leptos_mview::view;

fn style_on_component() {
    view! {
        Component style:color="white";
    }
}

fn class_on_component() {
    view! {
        Component class:red={true};
    }
}

fn prop_on_component() {
    view! {
        Component prop:value="1";
    }
}

#[component]
fn SpreadOnComponent() -> impl IntoView {
    #[allow(unused_variables)]
    let attrs = vec![
        ("class", "something"),
        ("data", "a"),
    ];
    view! {
        Component {..attrs};
    }
}

fn attr_on_element() {
    view! {
        input attr:class="no" type="text";
    }
}

fn clone_on_element() {
    let notcopy = String::new();
    view! {
        div {
            span clone:{notcopy} {
                {notcopy.clone()}
            }
        }
    }
}

fn sel_shorthand_on_components() {
    view! {
        Component.not-working #some-id;
    }
}

#[component]
fn Component() -> impl IntoView {
    view! {
        button;
    }
}

fn main() {}
