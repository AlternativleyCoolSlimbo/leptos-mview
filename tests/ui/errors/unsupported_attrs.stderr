error: directive `style` is not supported on components
 --> tests/ui/errors/unsupported_attrs.rs:6:19
  |
6 |         Component style:color="white";
  |                   ^^^^^^^^^^^

error: directive `class` is not supported on components
  --> tests/ui/errors/unsupported_attrs.rs:12:19
   |
12 |         Component class:red={true};
   |                   ^^^^^^^^^

error: directive `prop` is not supported on components
  --> tests/ui/errors/unsupported_attrs.rs:18:19
   |
18 |         Component prop:value="1";
   |                   ^^^^^^^^^^

error: spread attributes not supported on components
  --> tests/ui/errors/unsupported_attrs.rs:30:22
   |
30 |         Component {..attrs};
   |                      ^^^^^

error: directive `attr` is not supported on html elements
  --> tests/ui/errors/unsupported_attrs.rs:36:15
   |
36 |         input attr:class="no" type="text";
   |               ^^^^^^^^^^

error: directive `clone` is not supported on html elements
  --> tests/ui/errors/unsupported_attrs.rs:44:18
   |
44 |             span clone:{notcopy} {
   |                  ^^^^^^^^^^^^^^

error: class/id selector shorthand is not allowed on components
  --> tests/ui/errors/unsupported_attrs.rs:53:18
   |
53 |         Component.not-working #some-id;
   |                  ^^^^^^^^^^^^^^^^^^^^^
